xdsl/dialects/vector.py

```diff
@@ -282,6 +291,148 @@ def get(mask_operands: list[Operation | SSAValue]) -> CreateMaskOp:
         )
 
 
+def _get_mixed_position(
+    static_position: Sequence[int], dynamic_position: Sequence[SSAValue]
+) -> list[SSAValue | int]:
+    """
+    Returns a list of positions, represented as either an SSAValue or an int,
+    given the static and dynamic positions lists.
+    This function is used by the `vector.extract` and `vector.insert` verifiers.
+    """
+    next_dynamic_index = 0
+    result: list[SSAValue | int] = []
+    for pos in static_position:
+        assert isinstance(pos, int)
+        if pos == DYNAMIC_INDEX:
+            result.append(dynamic_position[next_dynamic_index])
+            next_dynamic_index += 1
+            continue
+        result.append(pos)
+    return result
+
+
+@irdl_op_definition
+class ExtractOp(IRDLOperation):
+    name = "vector.extract"
+
+    _T = Annotated[Attribute, ConstraintVar("T")]
+    _V = Annotated[VectorType[_T], ConstraintVar("V")]
```

Comment:
Can you make these ConstraintVarConstrs instead, please? We're migrating away from this API since it’s not officially supported by Pyright